sdcm/nemesis.py

```diff
@@ -1016,6 +1029,9 @@ def _kubernetes_wait_till_node_up_after_been_recreated(self, node, old_uid=None)
         self.monitoring_set.reconfigure_scylla_monitoring()
 
     def disrupt_terminate_and_replace_node(self):  # pylint: disable=invalid-name
+        if self._is_it_on_kubernetes():
+            raise UnsupportedNemesis(
```

Comment:
how about modyfing nemesis class, so it will select proper disrupt method when used on kubernetes?

Reply:
We already have separate nemesis for k8s. Not sure. Doing this change I just did "enough" to fix it.

Reply:
I'm telling about not adding unrelated to disruption logic - we should avoid responsibilities mixing. In that way we're making code more complex with full of 'if' statements.
Decision about selecting which nemesis we can run should be handled before running disrupt method. 
E.g. by filtering of available nemesis based if it's on k8s (or any other reason), possibly somewhere before testing is started. A class with one responsibility: filter out nemesis that cannot be run in current context (e.g. due k8s), and provide such a list for next step where we randomly choose one (or to Sisyphus to generate list).
This for example could be done by extracting disruption list creation (`call_random_disrupt_method` from `nemesis.py`) to some other class where we could include more filters, and reuse it wherever possible (code from mentioned method is duplicated in several places)

Reply:
I understand your point. but it sounds as "out of scope" for this change.
This change does "enough" small change to fix breakages existing now.
General refactor of the approach must be done separately. IMHO

Reply:
yes, might not be that easy.